### DISCUSSION TITLE
Maintaining coredns releases branches

### DISCUSSION THREAD

**Discussion by yashsingh74** [coredns/coredns#6150] [Ideas]
Currently, the coredns are cutting releases from the master branch and maintaining it. There isn't any scope for maintaining the older releases.  
The plan is to create a new branch whenever there is a release.
For eg... We can make a new branch for version v1.10 called **release/1.10**, which will be checked out from the master. The releases can then be cut from the same branch, along with any subsequent patches like **v1.10.0**, **v1.10.1**, and **v1.10.2**. 
This will make it easier for us to maintain the release branches properly. Similar processes are used by projects like **Kubernetes** and **containerd**. These new implementations will bring us closer to Kubernetes.

**Comment by chrisohaver** [comment]
This had been a conscious/intentional decision to only maintain a single minor release at a time. This was not because we don't know how to do it, rather the concern for the additional amount of work required to maintain and back-port fixes to previous minor releases.

Im not conceptually opposed to supporting multiple minor releases.  But before committing to it, we'd need to determine:

Will coredns maintainers and contributing community be able to dedicate the additional time to support this?
EOL policy - how far back do we support?

**Reply by yashsingh74** [reply]
As a small step, would it be a good idea to create a policy around patching older minor releases only with CVE fixes or critical fixes? Only support cherry-picking from maintained minor releases.
I am happy to come up with an initial draft of the policy for review and also help with contributing towards maintaining these patches.

Also, for EOL policy I think we can start with N-2 support.

**Reply by chrisohaver** [reply]
> I am happy to come up with an initial draft of the policy for review and also help with contributing towards maintaining these patches.

Thank you. I guess draft the policy in a new section in the project README.md ?  How/where do other projects document their support policy?

**Reply by yashsingh74** [reply]
Here for containerd - https://containerd.io/releases/
They mentioned LTS support and other release information.

**Reply by chrisohaver** [reply]
Ok. Add the policy to a new doc, RELEASES.md in coredns/coredns.

**Reply by yashsingh74** [reply]
Created a PR for the releases - https://github.com/coredns/coredns/pull/6224
